dbms/src/Storages/Page/V3/PageStorageImpl.cpp

```diff
@@ -403,14 +495,37 @@ void PageStorageImpl::registerExternalPagesCallbacks(const ExternalPageCallbacks
     assert(callbacks.scanner != nullptr);
     assert(callbacks.remover != nullptr);
     assert(callbacks.ns_id != MAX_NAMESPACE_ID);
-    assert(callbacks_container.count(callbacks.ns_id) == 0);
+    {
+#ifdef NDEBUG
+        // In product env, we assume that NamespaceId(TableID) will not be reuse,
+        // so that creating table don't require to be blocked by `callbacks_remove_mutex`.
```

Comment:
Are there any cases that we might also create, drop, recreate a new DTStore in a real production env?

Reply:
I meant although TiDB won't do that, is there any case TiFlash could happen?

Reply:
I think it will not happen unless we execute `drop table` by ch-client.

Or we need to acquire the lock on `callbacks_remove_mutex` that makes creating a table still being blocked by gc-ing DTFiles on "one table"